### PR TITLE
Add new task so we can publish to AzDO pipeline artifacts

### PR DESCRIPTION
**Pull request by JohnTortugo**
Closes: #1540 

This PR add a new MSBuild task `PushToAzureDevOpsArtifacts` to Tasks.Feed package. The new task has mostly the same interface as `PushToBlobFeed` and works mostly the same. Only exception is that the new task will publish to Azure Pipelines artifacts storage instead of feeds.

My plan is to have this task working side by side with PushToBlobFeed and then when the refactoring of publishing is completed we just remove the call to `PushToBlobFeed` in Publish.proj.

**Changes:**
- BuildManifestUtil.cs : utility methods for creating/managing build manifest.
- PushToAzureDevOpsArtifacts.cs : build task that is able to publish to azure artifacts and create build manifest.
- PushToBlobFeed.cs : just some change to use the new BuildManifestUtil class.

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/PushToBlobFeed.cs [hunk 1]
```diff
@@ -118,7 +114,9 @@ public async Task<bool> ExecuteAsync()
                         blobArtifacts = ConcatBlobArtifacts(blobArtifacts, symbolItems);
                     }
 
-                    CreateBuildManifest(blobArtifacts, packageArtifacts);
+                    BuildManifestUtil.CreateBuildManifest(Log, blobArtifacts, packageArtifacts,
```
Comment (jcagme) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/PushToBlobFeed.cs]: >BuildManifestUtil [](start = 20, length = 17)

Nit: One parameter per line

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/BuildManifestUtil.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        private static readonly char[] ManifestDataPairSeparators = { ';' };
+
+        public const string AssetsVirtualDir = "assets/";
+
+        public static void CreateBuildManifest(
+            TaskLoggingHelper log,
+            IEnumerable<BlobArtifactModel> blobArtifacts,
+            IEnumerable<PackageArtifactModel> packageArtifacts,
+            string assetManifestPath, string manifestRepoUri, string manifestBuildId,
+            string manifestBranch, string manifestCommit, string manifestBuildData)
```
Comment (jcagme) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/BuildManifestUtil.cs]: Nit: move to single line

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/PushToAzureDevOpsArtifacts.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                            Log.LogMessage(MessageImportance.High, $"##vso[artifact.upload containerfolder=BlobArtifacts;artifactname=BlobArtifacts]{symbolPath.ItemSpec}");
+                        }
+
+                        packageArtifacts = packageItems.Select(BuildManifestUtil.CreatePackageArtifactModel);
+                        blobArtifacts = symbolItems.Select(BuildManifestUtil.CreateBlobArtifactModel).Where(blob => blob != null);
+                    }
+
+                    BuildManifestUtil.CreateBuildManifest(Log, blobArtifacts, packageArtifacts,
+                        AssetManifestPath, ManifestRepoUri, ManifestBuildId,
+                        ManifestBranch, ManifestCommit, ManifestBuildData);
```
Comment (jcagme) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/PushToAzureDevOpsArtifacts.cs]: Nit: same splitting comment

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/PushToAzureDevOpsArtifacts.cs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                            .ToArray();
+
+                        foreach (var packagePath in packageItems)
+                        {
+                            Log.LogMessage(MessageImportance.High, $"##vso[artifact.upload containerfolder=PackageArtifacts;artifactname=PackageArtifacts]{packagePath.ItemSpec}");
+                        }
+
+                        foreach (var symbolPath in symbolItems)
+                        {
+                            Log.LogMessage(MessageImportance.High, $"##vso[artifact.upload containerfolder=BlobArtifacts;artifactname=BlobArtifacts]{symbolPath.ItemSpec}");
```
Comment (jcagme) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/PushToAzureDevOpsArtifacts.cs]: Nit: split long lines

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/BuildManifestUtil.cs (hunk 2 above)
Comment (JohnTortugo) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/BuildManifestUtil.cs]: all parameters? The line will have >300 chars.

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/BuildManifestUtil.cs [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+namespace Microsoft.DotNet.Build.Tasks.Feed
+{
+    public static class BuildManifestUtil
+    {
+        private static readonly char[] ManifestDataPairSeparators = { ';' };
+
+        public const string AssetsVirtualDir = "assets/";
+
+        public static void CreateBuildManifest(TaskLoggingHelper log, IEnumerable<BlobArtifactModel> blobArtifacts, IEnumerable<PackageArtifactModel> packageArtifacts, string assetManifestPath, string manifestRepoUri, string manifestBuildId, string manifestBranch, string manifestCommit, string manifestBuildData)
```
Comment (mmitche) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/BuildManifestUtil.cs]: Nit: Can you put the parameters on separate lines for easier readability?

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/BuildManifestUtil.cs (hunk 2 above)
Comment (mmitche) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/BuildManifestUtil.cs]: Yeah, shouldn't; be a single line there.

**Comment by mmitche** [comment]
Looks good.  So the next step would be to create the release pipeline for packages, then alter the publish package so that it calls this new task for the package case?

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/BuildManifestUtil.cs (hunk 2 above)
Comment (jcagme) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/BuildManifestUtil.cs]: What I meant was moving each to its own single line :) (me and English...)

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/BuildManifestUtil.cs (hunk 5 above)
Comment (jcagme) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/BuildManifestUtil.cs]: This is what I meant!

**Pull request merged**